src/main/java/org/digma/intellij/plugin/toolwindow/recentactivity/RecentActivityService.java

```diff
@@ -1,47 +1,389 @@
+                    processRecentActivityGoToSpanRequest(recentActivityGoToSpanRequest.getPayload(), project, editorService);
+                }
+                if (RECENT_ACTIVITY_GO_TO_TRACE.equalsIgnoreCase(reactMessageRequest.getAction())) {
+                    ActivityMonitor.getInstance(project).registerButtonClicked(traceButtonName,"recent-activity");
+                    RecentActivityGoToTraceRequest recentActivityGoToTraceRequest = parseJsonToObject(request, RecentActivityGoToTraceRequest.class);
+                    processRecentActivityGoToTraceRequest(recentActivityGoToTraceRequest, project);
+                }
+                if (RECENT_ACTIVITY_CLOSE_LIVE_VIEW.equalsIgnoreCase(reactMessageRequest.getAction())) {
+                    try {
+                        CloseLiveViewMessage closeLiveViewMessage = JsonUtils.stringToJavaRecord(request, CloseLiveViewMessage.class);
+                        RecentActivityService.getInstance(project).liveViewClosed(closeLiveViewMessage);
+                    } catch (Exception e) {
+                        //we can't miss the close message because then the live view will stay open.
+                        // close the live view even if there is an error parsing the message.
+                        Log.debugWithException(logger, project, e, "Exception while parsing CloseLiveViewMessage {}", e.getMessage());
+                        RecentActivityService.getInstance(project).liveViewClosed(null);
+                    }
+                }
+
+                callback.success("");
+                return true;
+            }
+
+            @Override
+            public void onQueryCanceled(CefBrowser browser, CefFrame frame, long queryId) {
+                super.onQueryCanceled(browser, frame, queryId);
+            }
+        }, true);
+
+        jbCefClient.getCefClient().addMessageRouter(msgRouter);
+
+        JPanel browserPanel = new JPanel();
+        browserPanel.setLayout(new BorderLayout());
+        browserPanel.add(jbCefBrowser.getComponent(), BorderLayout.CENTER);
+
+
+        JPanel jcefDigmaPanel = new JPanel();
+        jcefDigmaPanel.setLayout(new BorderLayout());
+        jcefDigmaPanel.add(browserPanel, BorderLayout.CENTER);
+
+        //sometimes there is a race condition on startup, a contextChange is fired before method info is available.
+        //calling environmentChanged will fix it
+        BackendConnectionMonitor backendConnectionMonitor = project.getService(BackendConnectionMonitor.class);
+        if (backendConnectionMonitor.isConnectionOk()) {
+            Backgroundable.ensureBackground(project, "change environment", () -> {
+                EnvironmentChanged publisher = project.getMessageBus().syncPublisher(EnvironmentChanged.ENVIRONMENT_CHANGED_TOPIC);
+                publisher.environmentChanged(analyticsService.getEnvironment().getCurrent(), true);
+            });
+        }
+
```

Comment:
This code was blindly copied from the other tool window and is not necessary here, you can see that also in the other tool window this code was commented out because its not necessary.
I mean the code that starts with the comment //sometimes there is a race condition...

Reply:
Removed the code